packages/neuron-ui/src/components/MultisigAddress/hooks.ts

```diff
@@ -365,3 +371,61 @@ export const useSubscription = ({
   }, [isLightClient, getAndSaveMultisigSyncProgress])
   return { multisigBanlances, multisigSyncProgress }
 }
+
+export const useCancelWithLightClient = () => {
+  const [isCloseWarningDialogShow, setIsCloseWarningDialogShow] = useState(false)
+  const onCancel = useCallback(() => {
+    setIsCloseWarningDialogShow(true)
+  }, [setIsCloseWarningDialogShow])
+  const onCancelCloseMultisigDialog = useCallback(() => {
+    setIsCloseWarningDialogShow(false)
+  }, [setIsCloseWarningDialogShow])
+  return {
+    isCloseWarningDialogShow,
+    onCancel,
+    onCancelCloseMultisigDialog,
+  }
+}
+
+export const useSetStartBlockNumber = ({
+  onUpdateConfig,
+}: {
+  onUpdateConfig: (v: Partial<MultisigEntity> & { id: number }) => Promise<void>
+}) => {
+  const [isSetStartBlockShown, setIsSetStartBlockShown] = useState(false)
+  const [editId, setEditId] = useState<number | undefined>()
+  const [address, setAddress] = useState<string | undefined>()
+  const [lastStartBlockNumber, setLastStartBlockNumber] = useState<number | undefined>()
+  const onConfirm = useCallback(
+    (startBlockNumber: number) => {
+      if (editId) {
+        return onUpdateConfig({
+          id: editId,
+          startBlockNumber,
+        }).then(() => {
+          setIsSetStartBlockShown(false)
+        })
+      }
+      return Promise.reject(new Error('The Edit multisig config is empty'))
+    },
+    [editId]
+  )
+  const openDialog = useCallback<React.MouseEventHandler<HTMLButtonElement>>(e => {
```

Comment:
It may not be a good practice to use HTML5 data attributes to extend an HTML button in React, since it is not common and could be hard to debug

```tsx
<button
  onClick={() => setSyncStartInfo({ id, address, startBlockNumber })
}
```

Reply:
I can not get the point. The `<React.MouseEventHandler<HTMLButtonElement>>` is defined for the function of e's type. And the button's `onClick` type is `<React.MouseEventHandler<HTMLButtonElement>>`